include/fiction/algorithms/simulation/sidb/random_layout_generator.hpp

```diff
@@ -0,0 +1,200 @@
+            {
+                // uniform distribution of [0,x_coordinate].
+                std::uniform_int_distribution<uint64_t> dist_x(0u, static_cast<uint64_t>(x_dimension));
+                // uniform distribution of [0,y_coordinate].
+                std::uniform_int_distribution<uint64_t> dist_y(0u, static_cast<uint64_t>(y_dimension));
+                // random integer from [0,x_coordinate] is selected.
+                const auto random_x_coordinate = dist_x(generator);
+                // random integer from [0,y_coordinate] is selected.
+                const auto random_y_coordinate = dist_y(generator);
+                // coordinate is constructed [x_coordinate, y_coordinate]
+                const auto random_coordinate = typename Lyt::coordinate({random_x_coordinate, random_y_coordinate});
+
+                bool constraint_violation_positive_sidbs = false;
+
+                if (parameter.prevent_positive_charges)
+                {
+                    // it checks if new coordinate is not closer than 2 cells (Euclidean distance).
+                    lyt.foreach_cell(
+                        [this, &lyt, &random_coordinate, &constraint_violation_positive_sidbs](const auto& c1)
+                        {
+                            if (euclidean_distance<Lyt>(lyt, c1, random_coordinate) < parameter.minimal_spacing)
+                            {
+                                constraint_violation_positive_sidbs = true;
+                            }
+                        });
+                }
+
+                if (!constraint_violation_positive_sidbs)
+                {
+                    lyt.assign_cell_type(random_coordinate, Lyt::cell_type::NORMAL);
+                }
+                attempt_counter += 1;
+            }
+
+            // it checks if new-found layout is identical to an already found layout (all_layouts).
+            uint64_t identical_layout_counter = 0;
+            for (const auto& old_lyt : previous_layouts)
+            {
+                old_lyt.foreach_cell(
+                    [&identical_layout_counter, lyt](const auto& cell_old) mutable
+                    {
+                        lyt.foreach_cell(
+                            [&identical_layout_counter, &cell_old](const auto& cell_new) mutable
+                            {
+                                if (cell_new == cell_old)
+                                {
+                                    identical_layout_counter += 1;
+                                }
+                            });
+                    });
+            }
+
+            // if all SiDBs are placed and the new layout is not a duplication, the layout is written to the
+            // std::ostream.
+            if (lyt.num_cells() == parameter.number_placed_sidbs && identical_layout_counter == 0)
+            {
+                layout = lyt;
+                write_sqd_layout(lyt, os);
```

Comment:
This feels like two concepts are mixed here. When I generate a random layout, I would not expect it to be written to a file, too. For that, I have another function that does that for me.